error_writer.go

```diff
@@ -40,8 +40,9 @@ type ErrorWriter struct {
 // NewErrorWriter constructs an ErrorWriter. To properly recognize supported
 // RPC Content-Types in net/http middleware, you must pass the same
 // HandlerOptions to NewErrorWriter and any wrapped Connect handlers.
+// Conditional options are ignored.
```

Comment:
Small nit: rather than assuming that the reader remembers all the available options, let's refer to the option directly:

```
Options supplied via [WithConditionalHandlerOptions] are ignored.
```

Also, this isn't accurate. We're not ignoring conditional options - we're applying them as though the Spec is unary. To actually ignore conditional options, we'd need to recursively flatten options (expanding out `WithOptions` and friends), then strip out the conditional options. Alternatively, we could flatten and apply _all_ options (ignoring the conditional func), so we're sure that we get the superset of all codecs.

@jhump and @emcfarlane - what do you want to do here?

Reply:
The conditional function is applied but the options the conditional has are skipped by:
```
	if spec.Procedure == "" {
		return // ignore empty specs
	}
```

Reply:
Ah, you're right. I totally missed that 🤦🏽‍♂️  That's ok with me, then.